Gems/Atom/Utils/Code/Include/Atom/Utils/AssetCollectionAsyncLoader.h

```diff
@@ -99,6 +99,8 @@ namespace AZ
 
         //! The key is the asset path
         AZStd::unordered_map<AZStd::string, Data::Asset<Data::AssetData>> m_readyAssets;
+        //! Assets that are queued for loading, waiting for OnAssetReady event.
+        AZStd::unordered_map<AZStd::string, Data::Asset<Data::AssetData>> m_notReadyAssets;
```

Comment:
Is this map necessary? I see data being stored here but nothing actually reads it.

Reply:
It's stored to avoid releasing the asset upon being queued for load. Need to store the reference somewhere to avoid ref counting to 0.